traitsui/qt4/drop_editor.py

```diff
@@ -56,6 +57,15 @@ def init(self, parent):
 
         self.control._qt4_editor = self
 
+    def dispose(self):
+        """ Disposes of the content of an editor.
+        """
+        if self.factory.readonly:
+            # enthought/traitsui#884
+            _BaseEditor.dispose(self)
```

Comment:
I wonder if it would be better to call `super(Editor, self).dispose()` without relying on what exactly that parent class is.

Reply:
If the design was right, we should have been able to just call `super().dispose()`. Unfortunately we can't. The `self.factory.readonly` branch in `init` does not use the `SimpleEditor.init`, so it would be wrong to call `super(Editor, self).dispose()` for this branch. If the `SimpleEditor.init` changes, its `SimpleEditor.dispose` will need to change, but that updated `SimpleEditor.dispose` may not be compatible with `DropEditor` any more.   I strongly doubt this editor should be subclassing the simple text editor. The purposes of these two editors look unrelated. It is possible that the subclassing was done "to reuse some code", causing unrelated components to be coupled. This problem is relevant for so so many editors and I'd in favour of decoupling them.

Reply:
Sorry, my comment wasn't clear. I meant that maybe we should use `super(Editor, self).dispose()` to call the `dispose` of what is now imported as `_BaseEditor` (the parent class of `.text_editor.SimpleEditor` aka `Editor`). From here it doesn't matter what the parent class of `.text_editor.SimpleEditor` is as long as its `dispose` method is called so I think we could do without the assumption here that the parent class is `_BaseEditor`.

As long as the parent class of `.text_editor.SimpleEditor` doesn't change, the behaviour of both statements should be equivalent.

Reply:
I am not sure I understand... This is not making assumption what the subclass of `traitsui.qt4.text_editor.SimpleEditor` is. The `DropEditor` should independently have subclassed `traitsui.qt4.editor.Editor` (which is base class for PyQt editors for Traits-based UIs) and have wanted to use call the `dispose` resolved on `traitsui.qt4.editor.Editor`. Currently `traitsui.qt4.editor.Editor.dispose` is not implemented, and it inherits the one from `traitsui.editor.Editor.dispose`. If `traitsui.qt4.editor.Editor.dispose` is later implemented, this particular branch in `DropEditor.dispose` would want to use that. What `traitsui.qt4.editor.text_editor.SimpleEditor.dispose` does should not matter because `traitsui.qt4.editor.text_editor.SimpleEditor.init` isn't used in this branch.

Reply:
Oh, okay. I misunderstood the intentions of the code. I thought the goal was to simply avoid `traitsui.qt4.text_editor.SimpleEditor` and to stick with its inheritance structure, whatever it would be (and importing `_BaseEditor` just because it happens to be the next class in that structure).

Since the goal is to call `traitsui.qt4.editor.Editor.dispose()` independently, the code is fine as it is 👍

Reply:
I see. I think I might have a comment in one of these PRs that says "skip this parent class ...". Maybe that was what caused the confusion. I may try to see if I can track that comment down and make it less confusing.